pkg/kube/roles.go

```diff
@@ -177,11 +188,11 @@ func UpdateUserRoles(kubeClient kubernetes.Interface, jxClient versioned.Interfa
 	return nil
 }
 
-func userRolesFor(userKind string, userName string, envRoles map[string]*v1.EnvironmentRoleBinding) []string {
+func userRolesFor(userKind string, userName string, userNamespace string, envRoles map[string]*v1.EnvironmentRoleBinding) []string {
 	answer := []string{}
 	for _, envRole := range envRoles {
 		for _, subject := range envRole.Spec.Subjects {
-			if subject.Kind == userKind && subject.Name == userName {
+			if subject.Kind == userKind && subject.Name == userName && subject.Namespace == userNamespace {
```

Comment:
we could have users in multiple namespaces. unlikely but possible - so lets at least check.